src/main/scala/beam/agentsim/infrastructure/parking/ParkingNetwork.scala

```diff
@@ -0,0 +1,86 @@
+package beam.agentsim.infrastructure.parking
+
+import beam.agentsim.Resource.ReleaseParkingStall
+import beam.agentsim.agents.vehicles.VehicleManager
+import beam.agentsim.infrastructure._
+import beam.utils.metrics.SimpleCounter
+import com.typesafe.scalalogging.LazyLogging
+import org.matsim.api.core.v01.Id
+
+abstract class ParkingNetwork[GEO: GeoLevel](
+  vehicleManagerId: Id[VehicleManager],
+  parkingZones: Map[Id[ParkingZoneId], ParkingZone[GEO]]
+) extends LazyLogging {
+
+  // Generic
+  protected val searchFunctions: Option[InfrastructureFunctions[_]]
+
+  // Core
+  protected var totalStallsInUse: Long = 0L
```

Comment:
I can look into the true usage a bit more, but we may need to be careful about parallelism here (and maybe this is already the case before the refactor). Locks might be needed around the usage of stalls